src/cmd/troubleshoot/image.go

```diff
@@ -192,83 +196,73 @@ func checkCustomModuleImagePullable(httpClient *http.Client, componentName strin
 }
 
 func imageAvailable(httpClient *http.Client, imageUrl string, apiToken string) error {
-	if statusCode, err := connectToDockerRegistry(httpClient, "HEAD", imageUrl, "Basic", apiToken); err != nil {
+	statusCode, err := connectToDockerRegistry(httpClient, imageUrl, apiToken)
+
+	if err != nil {
 		return fmt.Errorf("registry unreachable: %w", err)
-	} else {
-		if statusCode != http.StatusOK {
-			return fmt.Errorf("image not found (status code = %d)", statusCode)
-		}
+	} else if statusCode != http.StatusOK {
+		return fmt.Errorf("image not found (status code = %d)", statusCode)
 	}
+
 	return nil
 }
 
 func registryAvailable(httpClient *http.Client, registry string, apiToken string) error {
-	if statusCode, err := connectToDockerRegistry(httpClient, "HEAD", "https://"+registry, "Basic", apiToken); err != nil {
-		return fmt.Errorf("registry '%s' unreachable: %v", registry, err)
-	} else {
-		if statusCode != http.StatusOK {
-			return fmt.Errorf("registry '%s' unreachable (%d)", registry, statusCode)
-		}
-	}
-	return nil
-}
+	statusCode, err := connectToDockerRegistry(httpClient, registryUrl(registry), apiToken)
 
-func addProxy(troubleshootCtx *troubleshootContext) error {
-	proxyUrl := ""
-	if troubleshootCtx.dynakube.Spec.Proxy != nil {
-		if troubleshootCtx.dynakube.Spec.Proxy.Value != "" {
-			proxyUrl = troubleshootCtx.dynakube.Spec.Proxy.Value
-		} else if troubleshootCtx.dynakube.Spec.Proxy.ValueFrom != "" {
-			var err error
-			proxyUrl, err = kubeobjects.ExtractToken(&troubleshootCtx.proxySecret, dtclient.CustomProxySecretKey)
-			if err != nil {
-				return errorWithMessagef(err, "failed to extract proxy secret field")
-			}
-		}
-	}
-	if proxyUrl != "" {
-		p, err := url.Parse(proxyUrl)
-		if err != nil {
-			return errorWithMessagef(err, "could not parse proxy URL!")
-		}
-		t := troubleshootCtx.httpClient.Transport.(*http.Transport)
-		t.Proxy = http.ProxyURL(p)
-		logInfof("using  '%s' proxy to connect to the registry", p.Host)
+	if err != nil {
+		return fmt.Errorf("registry '%s' unreachable: %v", registry, err)
+	} else if statusCode != http.StatusOK {
```

Comment:
unnecessary else

Reply:
if i had written a second if statement, there would be no need for it, as it could be concatenated with an else. I.e. what advantage does a second if-statement have to an else branch that I should change the code that way?